packages/dds/tree/src/test/feature-libraries/indexing/treeIndex.spec.ts

```diff
@@ -0,0 +1,210 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert, fail } from "assert";
+
+import { getView } from "../../utils.js";
+import {
+	type FlexTreeNode,
+	AnchorTreeIndex,
+	flexTreeSlot,
+	isTreeValue,
+} from "../../../feature-libraries/index.js";
+import type { AnchorNode, FieldKey, ITreeSubscriptionCursor } from "../../../core/index.js";
+import { brand, disposeSymbol, getOrCreate } from "../../../util/index.js";
+import {
+	getOrCreateInnerNode,
+	SchemaFactory,
+	TreeViewConfiguration,
+	type TreeNode,
+} from "../../../simple-tree/index.js";
+// eslint-disable-next-line import/no-internal-modules
+import type { SchematizingSimpleTreeView } from "../../../shared-tree/schematizingTreeView.js";
+// eslint-disable-next-line import/no-internal-modules
+import { treeApi } from "../../../shared-tree/treeApi.js";
+// eslint-disable-next-line import/no-internal-modules
+import { getOrCreateNodeFromFlexTreeNode } from "../../../simple-tree/proxies.js";
+
+function readStringField(cursor: ITreeSubscriptionCursor, fieldKey: FieldKey): string {
+	cursor.enterField(fieldKey);
+	cursor.enterNode(0);
+	const { value } = cursor;
+	cursor.exitNode();
+	cursor.exitField();
+	assert(typeof value === "string");
+	return value;
+}
+
+describe.only("tree indexes", () => {
```

Comment:
we should have a clear-box test that checks that nodes that are GC'd after being removed are deleted from the index (triggered by "afterDestroy")